hummingbot/core/data_type/order_book_tracker.py

```diff
@@ -212,3 +220,48 @@ def snapshot(self) -> Dict[str, Tuple[pd.DataFrame, pd.DataFrame]]:
             except Exception:
                 self.logger().error("Unknown error. Retrying after 5 seconds.", exc_info=True)
                 await asyncio.sleep(5.0)
+
+    async def _emit_trade_event_loop(self):
```

Comment:
If the `_emit_trade_event_loop()` is a common function expected to be used by all tracker subclasses... it'd be better to start and stop it in the parent class's `start()` and `stop()` functions, and then require the subclasses to call the parent class's `start()` and `stop()` instead.